### PR TITLE
[#5157]fix: User can specify the krb5 conf file path for Kerberos enabled Hadoop catalog

### PR DESCRIPTION
**Pull request by tyoushinya**
### What changes were proposed in this pull request?

User can specify the krb5 conf file path for Kerberos enabled Hadoop catalog.

### Why are the changes needed?

Fix: #5157 

### Does this PR introduce _any_ user-facing change?

Add a property key java.security.krb5.conf for Kerberos enabled Hadoop catalog.

### How was this patch tested?

Create a Hadoop catalog for Kerberos enabled HDFS, you can put krb5.conf in any readable path of Gravatino host, and add the config java.security.krb5.conf=/X/krb5.conf , then the Hadoop catalog can be created succeed and all works well.

### CODE REVIEW

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java [hunk 1]
```diff
@@ -125,5 +139,13 @@ public int getFetchTimeoutSec() {
                   false /* immutable */,
                   60 /* defaultValue */,
                   false /* hidden */))
+          .put(
+              KRB5_CONF_KEY,
+              PropertyEntry.stringOptionalPropertyEntry(
+                  KRB5_CONF_KEY,
+                  "The Kerberos krb file for the catalog",
+                  false /* immutable */,
+                  "/etc/krb5.conf" /* defaultValue */,
```
Comment (yuqi1129) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java:148]: As the default value is `/etc/krb5.conf`, what if the file is not found?

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java [hunk 2]
```diff
@@ -63,6 +63,12 @@ public String login(String keytabFilePath) throws IOException {
     Preconditions.checkArgument(
         principalComponents.size() == 2, "The principal has the wrong format");
 
+    // Set Kerberos krb file
+    String krbFilePath = kerberosConfig.getKrb5Conf();
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(krbFilePath), "The Kerberos krb file can't be blank");
+    System.setProperty("java.security.krb5.conf", krbFilePath);
```
Comment (yuqi1129) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: Is this method class loader isolated? I mean what if I set `java.security.krb5.conf` in `catalogA`, does it take effect  in `catalogB`

File: docs/hadoop-catalog.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Besides the [common catalog properties](./gravitino-server-config.md#gravitino-c
...
 | `authentication.type`                              | The type of authentication for Hadoop catalog, currently we only support `kerberos`, `simple`.                                                                                                                                                                                                                 | `simple`        | No                                                          | 0.5.1            |
+| `java.security.krb5.conf`                          | Kerberos krb file for configuration of Kerberos.                                                                                                                                                                                                                                                              | /etc/krb5.conf  | No                                                          | 0.7.0-incubating |
```
Comment (yuqi1129) [inline docs/hadoop-catalog.md:35]: Let me think if there is a better name for it, `java.security.krb5.conf` is so long.

Comment (yuqi1129) [reply docs/hadoop-catalog.md:35]: what about `krb5.conf.path`?

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java (hunk 1 above)
Comment (tyoushinya) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java:148]: If the file is not exist, "Can't get Kerberos realm" error message will be shown. This behavior is same as before.

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java (hunk 2 above)
Comment (tyoushinya) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: The configuration is separated for each catalog, this mean the setting of catalogA can not affect catalogB.

**Comment by jerqi** [comment]
`krb5 conf` can't be config option of the catalog. `System.properties()` can't be isolated with the class loader.

File: docs/hadoop-catalog.md (hunk 3 above)
Comment (tyoushinya) [reply docs/hadoop-catalog.md:35]: LGTM about the name changing. That mean I should create a new PR for it, is it correct?

Comment (yuqi1129) [reply docs/hadoop-catalog.md:35]: > That mean I should create a new PR for it?

No, if it's okay, please do it in this PR.

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java (hunk 2 above)
Comment (yuqi1129) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: I mean if we set the value in `catalogA`, and in catalogB,  could we still get the value if `catalogB` does not set it?

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java (hunk 1 above)
Comment (yuqi1129) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosConfig.java:148]: OK

**Pull request closed without merging**